File: golang/stack/removing_stars_from_a_string.go
package main

func removeStars(s string) string {
	stack := []rune{}

	for _, ch := range s {
		if ch == '*' {
			if len(stack) > 0 {
				stack = stack[:len(stack)-1]
			}
		} else {
			stack = append(stack, ch)
		}
	}

	return string(stack)
}

File: java/src/main/java/com/sanedge/interview/stack/RemovingStarsFromAString.java
package com.sanedge.interview.stack;

import java.util.Stack;

public class RemovingStarsFromAString {
  public String removeStars(String s) {
    Stack<Character> stack = new Stack<>();

    for (char ch : s.toCharArray()) {
      if (ch == '*') {
        if (!stack.isEmpty()) {
          stack.pop();
        }
      } else {
        stack.push(ch);
      }
    }

    StringBuilder result = new StringBuilder();
    for (char ch : stack) {
      result.append(ch);
    }

    return result.toString();
  }

}
